Preparatory Doc Updates (Common, Browser Section, Other) (#3984)
ethereumjs/ethereumjs-monorepo: packages/common/README.md

## Browser

> We provide hybrid ESM/CJS builds for all our libraries. With the v10 breaking release round from Spring 2025 all libraries are "pure-JS" by default and we have eliminated all hard-wired WASM code. Additionally we have substantially lowered the bundle sizes, reduced the number of dependencies and cut out all usages of Node.js specific primities (like the Node.js event emitter).

Reviewer: ```suggestion
We provide hybrid ESM/CJS builds for all our libraries. With the v10 breaking release round from Spring 2025, all libraries are "pure-JS" by default and we have eliminated all hard-wired WASM code. Additionally we have substantially lowered the bundle sizes, reduced the number of dependencies, and cut out all usages of Node.js-specific primitives (like the Node.js event emitter).
```

## Events

> The `Common` class has a public property `events` which contains an `EventEmitter` (using [EventEmitter3](https://github.com/primus/eventemitter3)). Following events are emitted on which you can react within your code:

Reviewer: ```suggestion
The `Common` class has a public property `events` which contains an `EventEmitter` (using [EventEmitter3](https://github.com/primus/eventemitter3)). The following events are emitted on which you can react within your code:
```

To get an overview of the different parameters have a look at one of the chain configurations in the `chains.ts` configuration
file, or to the `Chain` type in [./src/types.ts](./src/types.ts).

### Working with Private/Custom Chains

> Starting with the `v10` release series using custom chain configurations has been simplified and consolidated in a single API `createCustomCommon()`. This constructor can be both used to make simple chain ID adjustments and keep the rest of the config conforming to a given "base chain":

Reviewer: ```suggestion
Starting with the `v10` release series, using custom chain configurations has been simplified and consolidated in a single API `createCustomCommon()`. This constructor can be both used to make simple chain ID adjustments and keep the rest of the config conforming to a given "base chain":
```

```ts
import { createCustomCommon, Mainnet } from '@ethereumjs/common'
 
createCustomCommon({chainId: 123}, Mainnet)
```

See the `Tx` library [README](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/tx) for how to use such a `Common` instance in the context of sending txs to L2 networks.

> Beyond it is possible to customize to a fully custom chain by passing in a complete configuration object as first parameter:

Reviewer: ```suggestion
Furthermore, it is possible to further customize a `Common` instance with a fully custom chain by passing in a complete configuration object as the first parameter:
```

- `muirGlacier` (`Hardfork.MuirGlacier`)
- `berlin` (`Hardfork.Berlin`) (since `v2.2.0`)
- `london` (`Hardfork.London`) (since `v2.4.0`)
- `merge` (`Hardfork.Merge`) (since `v2.5.0`)
- `shanghai` (`Hardfork.Shanghai`) (since `v3.1.0`)
- `cancun` (`Hardfork.Cancun`) (since `v4.2.0`)
- `prague` (`Hardfork.Prague`) (`DEFAULT_HARDFORK`) (since `v10`)

### Future Hardforks

> The next upcoming HF `Hardfork.Osaka` is currently not yet supported by this library.

Reviewer: ```suggestion
The next upcoming HF, `Hardfork.Osaka`, is currently not yet supported by this library.
```
